cli/cmd/init.go

```diff
@@ -301,49 +301,50 @@ func validateConfigurationIntegirty(ctx context.Context, slice string, happyClie
 	logrus.Debug("Scheduling validateConfigurationIntegirty()")
 	return func() error {
 		logrus.Debug("Running validateConfigurationIntegirty()")
-		// These services are configured in docker-compose.yml, and have their containers built
-		availableServices, err := happyClient.ArtifactBuilder.GetServices(ctx)
+
+		// Happy configuration is spread across these files:
+		// * .happy/config.json defines environments, specifies services, slices, features and tasks
+		// * docker-compose.yml defines services and their build configuration
+		// * terraform code from .happy/terraform/envs/<ENVNAME>/*.tf references services and their settings
+
+		// All services referenced through TF code must be present in config.json. All services listed in config.json
+		// must be declared in docker-compose.yml and have a build section.
+
+		// These services are configured in docker-compose.yml
+		composeServices, err := happyClient.ArtifactBuilder.GetServices(ctx)
 		if err != nil {
 			return errors.Wrap(err, "unable to get available services")
 		}
 
-		// NOTE: availableServices will only contain the services that are a part of the slice.
-		// That means we have to iterate over those first to make sure they are all in the config.json and
-		// not the other way around.
+		// ConfigServices are configured in config.json, and are a subset of the services in docker-compose.yml.
+		// Every service from config.json must be present in docker-compose.yml and must have a build section.
 		configServices := happyClient.HappyConfig.GetServices()
 		ss := sets.NewStringSet().Add(configServices...)
-		for serviceName, service := range availableServices {
-			// ignore services that don't have a build section
-			// as these are not deployable services
-			if service.Build == nil {
-				continue
-			}
-			ok := ss.ContainsElement(serviceName)
+		for _, serviceName := range configServices {
+			service, ok := composeServices[serviceName]
```

Comment:
thinking...

I had this loop like this for a different bug but trying to remember what it was

Reply:
ooo slices. that is why. This approach doesn't work when you are working with slices. See my comment:
```

// NOTE: availableServices will only contain the services that are a part of the slice.
		// That means we have to iterate over those first to make sure they are all in the config.json and
		// not the other way around.
```

Reply:
To test this out, I would borrow one of the single-cell apps and make sure your validation works as expected when using their slices

Reply:
@jgadling issue is the exact opposite issue -- for a non-slice setup a service that has a build section in docker-compose but is not a happy service, validation fails.

Reply:
I don't know what this means, we're not using slices, but I'm inferring from your comment that I can work around the issue I'm having by removing the (unnecessary) "build" section from the second service in my docker-compose file.